test-frame-metrics-collector/src/main/java/io/skodjob/testframe/MetricsCollector.java

```diff
@@ -397,6 +411,11 @@ public final synchronized ArrayList<Double> waitForSpecificMetricAndCollect(Patt
      */
     protected String collectMetrics(String metricsPodIp, String podName)
         throws InterruptedException, ExecutionException, IOException {
+
+        if (this.deployScraperPod) {
```

Comment:
Hmm when I'm thinking about it, would it make sense to have those `deploy` and `delete` methods public and instead of deploying it (deleting it) on every call of `collectMetrics`, users can deploy it and delete it anytime they want. WDYT?

Reply:
I was thinking about it as well, but I think that it should be one time pod for collect metrics and then be killed to do not affects other steps in tests like counting pods etc... so let's keep it just for this purpose as one time pod.

Reply:
Agree that in case that it's small image and single Pod, it's not a big problem. Thanks